### ISSUE TITLE
Rtags on Mac OS X for Ubuntu project

### ISSUE THREAD

**Issue by keeyim** [Andersbakken/rtags#1058]
### Please mark appropriate
 - [x] rtags (rdm/rc/rp)
 - Emacs Lisp
   - [ ] ac-rtags.el
   - [ ] company-rtags.el
   - [ ] helm-rtags.el
   - [ ] flycheck-rtags.el
   - [ ] ivy-rtags.el
   - [ ] rtags.el

### Use case
I am trying out rtags to help with navigating the code in a large project, specifically  [https://github.com/ceph/ceph].

My workflow is to browse/search/edit the code on my Mac and do the compilation/execution in an Ubuntu VM.

What I had in mind was to run the `rdm` daemon on the Mac, since that is where I do my code browsing (in Atom).

### Where I'm stuck
I need to generate the `compile_commands.json` for the ceph project. That project uses cmake, so I'd like to use `cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1`, however the ceph project builds only for Linux flavors (Ubuntu in my case) and not for Mac OS X.

I can go into my Ubuntu VM and run the cmake command, however the compile commands generated would be for Ubuntu (with wrong paths) and not for Mac.

**So my question is this: how do I create an appropriate `compile_commands.json` for the Mac side so that I can run rtags there?**

Apologies if this information is out there somewhere; I couldn't find it. I did find the following somewhat similar issues, but none were exactly what I am looking for: #100, #594, #735

Any and all help would be appreciated. Thanks!

**Comment by Andersbakken** [comment]
I think the best way likely is to use cmake to generate the
compile_commands.json on ubuntu and then run a little sed command to modify
the paths to be correct.

Anders

**Comment by keeyim** [comment]
Thanks Anders. I will give that a shot and report back.

**Comment by keeyim** [comment]
I finally got this working.

In my Ubuntu VM, I created an rtags-specific build dir for my project (so as not to mess up my linux build), and used cmake to generate compile_commands.json. I used sed to change my VM-specific paths to the correct ones on the Mac side.

The most painful part was related to the project I was tagging, since I had to hack around a bit to work around hardcoded assumptions in its code and cmake infrastructure about running on a linux system.

I kicked off loading my project with `rc -J` and used `/usr/local/var/log/rtag.log` to figure out where the compilation was failing. After iterating on this for a while, I was able to successfully load the files in my project.

I've now got `rtags` hooked into my Mac's launchd and am using Atom with the atomic-rtags package. Nice!

Thanks for the help.
